tests/phpunit/integration/Core/User_Input/User_InputTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Core\User_Input\User_InputTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\User_Input
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\User_Input;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\User_Input\User_Input;
+use Google\Site_Kit\Core\User_Input\Site_Specific_Questions;
+use Google\Site_Kit\Core\User_Input\User_Specific_Questions;
+use Google\Site_Kit\Tests\TestCase;
+
+class User_InputTest extends TestCase {
+
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	* Context object.
+	*
+	* @var User_Input
+	*/
+	private $user_input;
+
+	/**
+	* User ID.
+	*
+	* @var int
+	*/
+	private $user_id;
+
+	/**
+	 * Settings with empty default values.
+	 *
+	 * @var array|ArrayAccess
+	 */
+	private static $empty_settings = array(
+		'purpose'       => array(
+			'scope'  => 'site',
+			'values' => array(),
+		),
+		'postFrequency' => array(
+			'scope'  => 'user',
+			'values' => array(),
+		),
+		'goals'         => array(
+			'scope'  => 'user',
+			'values' => array(),
+		),
+	);
+
+	/**
+	 * Settings with dummy values.
+	 *
+	 * @var array|ArrayAccess
+	 */
+	private static $dummy_settings = array(
```

Comment:
Some of the language here goes against Google's standards. See https://github.com/google/site-kit-wp/issues/3031

Additionally, are these expected to be valid or invalid? Perhaps something like `valid_non_empty_settings` would be better?